gremlin-go/README.md

```diff
@@ -65,91 +58,222 @@ import (
+
 # Go Gremlin Language Variant
 
 [Apache TinkerPop™][tk] is a graph computing framework for both graph databases (OLTP) and graph analytic systems
 (OLAP). [Gremlin][gremlin] is the graph traversal language of TinkerPop. It can be described as a functional,
 data-flow language that enables users to succinctly express complex traversals on (or queries of) their application's
 property graph.
 
-Gremlin-Go implements Gremlin within the Go language and can be used on any Go runtime greater than v1.17. Go's syntax 
-has the same constructs as Java including "dot notation" for function chaining (a.b.c), round bracket function arguments
-(a(b,c)), and support for global namespaces (a(b()) vs a(__.b())). One important distinction with Go and Java is that 
-the functions are capitalized, as is required to export functions is Go. As such, anyone familiar with Gremlin-Java 
-will immediately be able to work with Gremlin-Go.
+Gremlin-Go implements Gremlin within the Go language and can be used on any Go runtime greater than v1.17. One important distinction with Go and Java is that 
+the functions are capitalized, as is required to export functions is Go. 
 
 Gremlin-Go is designed to connect to a "server" that is hosting a TinkerPop-enabled graph system. That "server"
 could be [Gremlin Server][gs] or a [remote Gremlin provider][rgp] that exposes protocols by which Gremlin-Go
 can connect.
 
 A typical connection to a server running on "localhost" that supports the Gremlin Server protocol using websockets
 looks like this:
-<!--
-TODO: Add Go code example of connection to server.
--->
+```go
+package main
+
+import (
+	"fmt"
+	"github.com/lyndonb-bq/tinkerpop/gremlin-go/driver"
+)
+
+func main() {
+	// Creating the connection to the server.
+	driverRemoteConnection, err := gremlingo.NewDriverRemoteConnection("localhost", 8182)
+	// Handle error
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	// Create an anonymous traversal source with remote
+	g := gremlingo.Traversal_().WithRemote(driverRemoteConnection)
+}
+```
 
 Once "g" has been created using a connection, it is then possible to start writing Gremlin traversals to query the
 remote graph:
-<!--
-TODO: Add Go code example of a Gremlin traversal query.
--->
+```go
+package main
+
+import (
+	"fmt"
+	"github.com/lyndonb-bq/tinkerpop/gremlin-go/driver"
+)
+
+func main() {
+	// Creating the connection to the server.
+	driverRemoteConnection, err := gremlingo.NewDriverRemoteConnection("localhost", 8182)
+	// Handle error
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	// Create an anonymous traversal source with remote
+	g := gremlingo.Traversal_().WithRemote(driverRemoteConnection)
 
-# The following material is currently Work-in-progress: 
+	// Add a vertex with properties to the graph with the terminal step Iterate()
+	_, promise, err := g.AddV("gremlin").Property("language", "go").Iterate()
+	if err != nil {
+		fmt.Println(err)
+		return 
+	}
+	<-promise
```

Comment:
This part with the promise is kind of confusing. Perhaps add a bit of commenting? It is only not confusing to someone that is both well-versed in both Gremlin and Go.

Reply:
Oh I had a comment in the subsequent block but forgot to add it here. Would `// Wait for all steps to finish execution` suffice?

Reply:
I think something quick explaining what type the promise is and that we're using `channels` here would be good so someone can google it if they arent well-versed.